### PR TITLE
Add crop support for Pam's Harvestcraft and Roots herbs

### PR DESCRIPTION
**Pull request by gchpaco**
This adds support for Pam's many crops and Roots herbs to the Harvest Moon ritual.  Rather than enumerate all the Pam's crops, it uses reflection to get a list from Harvestcraft itself.

### CODE REVIEW

File: src/main/java/WayofTime/bloodmagic/ritual/harvest/HarvestHandlerPlantable.java [hunk 1]
```diff
@@ -98,4 +113,37 @@ private static void addThirdPartyCrop(String modid, String regName, int matureMe
...
+            String className = "com.pam.harvestcraft.blocks.CropRegistry";
+            Class<?> registry = ReflectionHelper.getClass(loader, className);
+            Field names = ReflectionHelper.findField(registry, "cropNames");
+            Method getCrop = registry.getMethod("getCrop", String.class);
+            for (String name : (String[])names.get(null)) {
+                Block crop = (Block) getCrop.invoke(null, name);
+                HarvestRegistry.registerStandardCrop(crop, 3);
+            }
+        } catch (NoSuchMethodException e) {
+            FMLLog.warning("HarvestCraft integration cancelled; unable to find crop name mapper");
```
Comment (TehNut) [inline src/main/java/WayofTime/bloodmagic/ritual/harvest/HarvestHandlerPlantable.java]: I would rather these used `BloodMagic.instance.getLogger()`

**Review by TehNut — CHANGES_REQUESTED** [review]

**Comment by TehNut** [comment]
Do all Pam's crops really mature at an age of 3? I thought they went up to 7.

Since you're already reflecting all the things and Pam uses BlockCrops, you could probably cast to that and call `getMaxAge()`.

**Comment by gchpaco** [comment]
I've used the new logger, and called `getMaxAge` per your thoughts.  The JavaDoc situation for Forge 1.11 is not the best, and I didn't actually realize `getMaxAge` existed.

Pam crops do actually mature at 3 in the version of the source code we have; but because that version is not in any way current it might have changed since then.  I suppose I could have decompiled it, but this project didn't seem merit that kind of detail.

**Comment by TehNut** [comment]
>Pam crops do actually mature at 3 in the version of the source code we have

Huh, didn't know that. I don't use Pam's so...

**Review by TehNut — APPROVED** [review]

**Comment by TehNut** [comment]
Thanks!

**Pull request merged**